### PR TITLE
Support PHP 8

### PR DESCRIPTION
**Pull request by edudobay**
Fixes the issues raised in https://github.com/jakubkulhan/bunny/pull/101.

There's one issue about dropping PHP 7.0 support:

- Add `void` return type to PHPUnit `setUp` methods: this makes the syntax incompatible with PHP 7.0

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -25,13 +25,13 @@
...
     "require": {
-        "php": "~7.0",
+        "php": "^7.1 || ^8.0",
         "psr/log": "~1.0",
         "react/event-loop": "^1.0 || ^0.5 || ^0.4",
         "react/promise": "~2.2"
     },
     "require-dev": {
-        "phpunit/phpunit": "~6.5"
+        "phpunit/phpunit": "^6.5 || ^7.5 || ^8.5 || ^9.3"
```
Comment (standa) [inline composer.json:34]:
```suggestion
        "phpunit/phpunit": "^8.5 || ^9.3"
```

@edudobay I think only these two make sense for `"php": "^7.1 || ^8.0",`: https://phpunit.de/supported-versions.html

Comment (edudobay) [reply composer.json:34]: @standa If PHP 7.1 is still to be supported, I think we need PHPUnit 7 too; Composer won't let us install PHPUnit 8 with PHP 7.1.

Comment (edudobay) [reply composer.json]: I've dropped PHPUnit 6 and pushed a new commit to fix PHP 8 build. What do you think @standa?

**Comment by edudobay** [comment]
Trying to fix these SSL issues here! It's hard to know what exactly went wrong, so I'm working on making local SSL testing easier as well — I had a hard time getting the SSL tests to run locally.

**Comment by standa** [comment]
@edudobay I tested this PR on our production code with php 8.0 today and it works like a charm!

**Comment by edudobay** [comment]
Nice to hear that, @standa! I just have to solve this SSL issue in the CI builds — I'll try to reproduce the CI environment on the next few days to see how I can fix this.

**Comment by WyriHaximus** [comment]
@standa awesome! 

Planning to release `v0.4.4` to tomorrow

@edudobay let me know when this is ready, it looks like a breaking change to assigned `v0.5.0` to this PR

**Comment by edudobay** [comment]
@WyriHaximus Yes, now this is ready :) And in fact there's a breaking change, so it also seems to me that it would be better suited for the 0.5.0 release.

**Comment by WyriHaximus** [comment]
Ok just release `v0.4.4`, going to merge this soon and then add some additional workflows before tagging `v0.5.0`. Want to start keeping a better release message from here on out.

**Review by WyriHaximus — APPROVED** [review]
👏 👏 👏 👏  Well done! And thank you for putting the time into this 👍

**Pull request merged**